crates/bcr-wallet-core/src/app.rs

```diff
@@ -279,6 +279,18 @@ pub async fn wallet_reclaim_funds(idx: usize) -> Result<WalletBalance> {
     Ok(balance)
 }
 
+pub async fn wallet_redeem_credit(idx: usize) -> Result<cashu::Amount> {
+    tracing::debug!("wallet_redeem({idx})");
+    let wallet: Rc<ProductionWallet> =
+        APP_STATE.with_borrow(|state| -> Result<Rc<ProductionWallet>> {
```

Comment:
This pattern is repeated several times in this file - could this maybe be a fallible function like `get_wallet_by_idx`, or something?